app/controllers/trainees/degrees_controller.rb

```diff
@@ -2,62 +2,73 @@
 
 module Trainees
   class DegreesController < ApplicationController
-    def new
-      authorize trainee
-      @degree = trainee.degrees.build(locale_code: params[:locale_code])
-    end
+    before_action :authorize_trainee
+    before_action :set_trainee_form
 
-    def edit
-      authorize trainee
-      @degree = trainee.degrees.from_param(params[:id])
+    def new
+      @degree = @trainee_form.build_degree(locale_code: params[:locale_code])
```

Comment:
If we have `@degrees_form` instread `@trainee_form`, then `@degrees_form.build` would read nicely IMHO.